core/src/main/scala/org/apache/spark/deploy/rest/SubmitRestProtocolRequest.scala

```diff
@@ -61,7 +61,7 @@ private[rest] class CreateSubmissionRequest extends SubmitRestProtocolRequest {
     assertProperty[Boolean](key, "boolean", _.toBoolean)
 
   private def assertPropertyIsNumeric(key: String): Unit =
-    assertProperty[Int](key, "numeric", _.toInt)
+    assertProperty[Double](key, "numeric", _.toDouble)
```

Comment:
This is actually for supporting fractional cores when providing --driver-cores.
Cores in Spark are all integers, but it becomes quite limited when you want launch multiple jobs as each job minimumally has to take at least 1 core (2 in case of fine-grain mode). Making driver cores double you can then specifiy --driver-cores 0.1 and it will just use 10% of a single core share.